src/misc/cache/lv_cache_entry.c

```diff
@@ -169,17 +177,47 @@ void lv_cache_entry_init(lv_cache_entry_t * entry, const lv_cache_t * cache, con
     entry->cache = cache;
     entry->node_size = node_size;
     entry->ref_cnt = 0;
-    entry->is_invalid = false;
+    entry->flags = 0;
 }
 
 void lv_cache_entry_delete(lv_cache_entry_t * entry)
 {
     LV_ASSERT_NULL(entry);
 
+    if(entry->flags & LV_CACHE_ENTRY_FLAG_DISABLE_DELETE) {
+        return;
+    }
+
     void * data = lv_cache_entry_get_data(entry);
     lv_free(data);
 }
 
+void lv_cache_entry_set_class_flag(lv_cache_entry_t * entry, bool value)
+{
+
+    LV_ASSERT_NULL(entry);
+    if(value) {
+        entry->flags |= LV_CACHE_CLASS_FLAG;
+    }
+    else {
+        entry->flags &= ~LV_CACHE_CLASS_FLAG;
+    }
+}
+
+bool lv_cache_entry_get_class_flag(lv_cache_entry_t * entry)
```

Comment:
What about having a flag API similar to `lv_obj` with `lv_cache_entry_add/remove/has_flag`?